samcli/commands/init/init_templates.py

```diff
@@ -102,6 +106,17 @@ def clone_templates_repo(self):
                 if expected_previous_clone_local_path.exists():
                     self._git_repo.local_path = expected_previous_clone_local_path
 
+    def _check_upsert_templates(self, shared_dir, cloned_folder_name):
+        cache_dir = shared_dir / cloned_folder_name
+        git_executable = self._git_repo.get_git_executable()
+        command = [git_executable, "rev-parse", "--verify", "HEAD"]
```

Comment:
Something to keep in mind when using capabilities is to ensure when they were released so that we don't inadvertently create a min version requirement for git.

However, for this particular command I do not believe it's recent.

Reply:
This is part of the reason why we don't fail here if we fail to run the command. We'll just keep going with existing behaviour.